File: QA.Engine.Administration.WebApp.Core/ClientApp/models/WidgetViewModel.ts

module Models {

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]
    
    // More info: http://frhagn.github.io/Typewriter/
    
    
    export class WidgetViewModel {
        
        // ID
        public id: number = 0;
        // ISARCHIVE
        public isArchive: boolean = false;
        // PARENTID
        public parentId: number = null;
        // ALIAS
        public alias: string = null;
        // TITLE
        public title: string = null;
        // ZONENAME
        public zoneName: string = null;
        // EXTENSIONID
        public extensionId: number = null;
        // INDEXORDER
        public indexOrder: number = null;
        // ISVISIBLE
        public isVisible: boolean = null;
        // VERSIONOFID
        public versionOfId: number = null;
        // PUBLISHED
        public published: boolean = false;
        // DISCRIMINATORID
        public discriminatorId: number = 0;
        // CHILDREN
        public children: WidgetViewModel[] = [];
        // DISCRIMINATOR
        public discriminator: DiscriminatorViewModel = null;
        // REGIONS
        public regions: RegionViewModel[] = [];
        // HASCHILDREN
        public hasChildren: boolean = false;
        // HASREGIONS
        public hasRegions: boolean = false;
    }
}

File: QA.Engine.Administration.WebApp.Core/Models/WidgetViewModel.cs
using System.Collections.Generic;
using System.Linq;

namespace QA.Engine.Administration.WebApp.Core.Models
{
    public class WidgetViewModel
    {
        public int Id { get; set; }
        public bool IsArchive { get; set; }
        public int? ParentId { get; set; }
        public string Alias { get; set; }
        public string Title { get; set; }
        public string ZoneName { get; set; }
        public int? ExtensionId { get; set; }
        public int? IndexOrder { get; set; }
        public bool? IsVisible { get; set; }
        public int? VersionOfId { get; set; }
        public bool Published { get; set; }
        public string Discriminator { get; set; }
        public int DiscriminatorId { get; set; }
        public string DiscriminatorTitle { get; set; }
        public string IconUrl { get; set; }
        public List<WidgetViewModel> Children { get; set; }
        public List<RegionViewModel> Regions { get; set; }
        public bool HasChildren { get { return Children?.Any() ?? false; } }
        public bool HasRegions { get { return Regions?.Any() ?? false; } }
    }
}
